skada/deep/_adversarial.py

```diff
@@ -370,6 +369,148 @@ def CDAN(
     return net
 
 
+class MDDLoss(BaseDALoss):
+    """Loss MDD.
+
+    This loss tries to minimize the disparity discrepancy between
+    the source and target domains. The discrepancy is estimated
+    through adversarial training of three networks: an encoder,
+    a task network and a discriminator.
+
+    See [35]_ for details.
+
+    Parameters
+    ----------
+    domain_criterion : torch criterion (class)
+        The criterion (loss) used to compute the
+        MDD loss. If None, a CrossEntropyLoss is used.
+    gamma : float (default=4.0)
+        Margin parameter following [35]_
+
+    References
+    ----------
+    .. [35] Yuchen Zhang et. al. Bridging Theory and Algorithm
+            for Domain Adaptation. In International Conference on
+            Machine Learning, 2019.
+    """
+
+    def __init__(self, domain_criterion, gamma=4.0):
+        super().__init__()
+        if domain_criterion is None:
```

Comment:
I know you want to follow the DANN model but it's weird to call them domain_criterion. You can call it disc_criterion or as you want.

Reply:
Done